lua/mapvote/sv_mapvote.lua

```diff
@@ -87,127 +87,74 @@ local function mapVoteOver( autoGamemode, callback )
 
     local map = MapVote.CurrentMaps[winner]
 
-    local gamemode = nil
-
-    if autoGamemode then
-        -- check if map matches a gamemode's map pattern
-        for _, gm in pairs( engine.GetGamemodes() ) do
-            -- ignore empty patterns
-            if gm.maps and gm.maps ~= "" then
-                -- patterns are separated by "|"
-                for _, pattern in pairs( string.Split( gm.maps, "|" ) ) do
-                    if string.match( map, pattern ) then
-                        gamemode = gm.name
-                        break
-                    end
-                end
-            end
-        end
-    else
-        print( "not enabled" )
-    end
-
     timer.Simple( 4, function()
-        if ( hook.Run( "MapVoteChange", map ) ~= false ) then
-            if ( callback ) then
-                callback( map )
-            else
-                -- if map requires another gamemode then switch to it
-                if ( gamemode and gamemode ~= engine.ActiveGamemode() ) then
-                    RunConsoleCommand( "gamemode", gamemode )
-                end
-                RunConsoleCommand( "changelevel", map )
-            end
+        if hook.Run("MapVoteChange", map) == false then return end
+        if callback then
+            callback( map )
         end
+
+        RunConsoleCommand( "changelevel", map )
     end )
 end
 
-function MapVote.Start( length, current, limit, prefix, callback )
-    current = current or MapVote.Config.AllowCurrentMap or false
-    length = length or MapVote.Config.TimeLimit or 28
-    limit = limit or MapVote.Config.MapLimit or 24
-    prefix = prefix or MapVote.Config.MapPrefixes
-    local cooldown = MapVote.Config.EnableCooldown or MapVote.Config.EnableCooldown == nil and true
-    local autoGamemode = autoGamemode or MapVote.Config.AutoGamemode or MapVote.Config.AutoGamemode == nil and true
+local function isMapAllowed( m )
```

Comment:
single letter variable name is quite vague, something like `map` might be better